File: tests/code/corpus/langs/julia/shapes.jl
using LinearAlgebra
import Base: show

abstract type Shape end

struct Circle <: Shape
    radius::Float64
end

struct Point
    x::Float64
end

function area(c::Circle)
    return pi * c.radius * c.radius
end

function total_area(shapes)
    total = 0.0
    for shape in shapes
        total += area(shape)
    end
    return total
end

make_circle(radius) = Circle(radius)

File: tests/code/corpus/langs/dart/shapes.dart
import 'dart:math';

abstract class Shape {
  double area();
}

class Circle extends Shape {
  final double radius;

  Circle(this.radius);

  double area() {
    return pi * radius * radius;
  }
}

mixin Loggable {
  void log() {}
}

double totalArea(List<Shape> shapes) {
  double total = 0.0;
  for (final shape in shapes) {
    total += shape.area();
  }
  return total;
}

Circle makeCircle(double radius) {
  return Circle(radius);
}
